app/uk/gov/hmrc/ctcguaranteebalancerouter/connectors/EISConnector.scala

```diff
@@ -153,16 +170,27 @@ class EISConnectorImpl(
       }
     }
 
+  private def deriveErrorFromResponseMessage(response: EISResponse): ConnectorError =
+    (response.containsInvalidGRN, response.invalidAccessCode) match {
+      case (true, _) =>
+        ConnectorError.GrnNotFound
+      case (_, true) =>
+        ConnectorError.InvalidAccessCode
+      case _ => ConnectorError.Unexpected("Unexpected response from EIS", None)
+    }
```

Comment:
Not your fault, but the fact we have to do this is shocking...

Reply:
I know right!